`unused-lifetimes` lint duplicates "parameter is never used" error

The following code:

```rust
#![warn(unused_lifetimes)]
struct Foo<'a>;

fn main() {}

```

gives the following output:

```
arning: lifetime parameter `'a` never used
 --> unused_lifetime.rs:2:12
  |
2 | struct Foo<'a>;
  |           -^^- help: elide the unused lifetime
  |
note: the lint level is defined here
 --> unused_lifetime.rs:1:9
  |
1 | #![warn(unused_lifetimes)]
  |         ^^^^^^^^^^^^^^^^

error[E0392]: parameter `'a` is never used
 --> unused_lifetime.rs:2:12
  |
2 | struct Foo<'a>;
  |            ^^ unused parameter
  |
  = help: consider removing `'a`, referring to it in a field, or using a marker such as `std::marker::PhantomData`

error: aborting due to previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0392`.
```

The "unused lifetimes" warning is redundant - not using a lifetime parameter is a hard error, so the warning gives no additional information.

--- a/tests/ui/single-use-lifetime/dedup.rs
+++ b/tests/ui/single-use-lifetime/dedup.rs
@@ -1,0 +1,9 @@
+// Check that `unused_lifetimes` lint doesn't duplicate a "parameter is never used" error.
+// Fixed in <https://github.com/rust-lang/rust/pull/96833>.
+// Issue: <https://github.com/rust-lang/rust/issues/72587>.
+
+#![warn(unused_lifetimes)]
+struct Foo<'a>;
+//~^ ERROR parameter `'a` is never used
+
+fn main() {}

--- a/tests/ui/single-use-lifetime/dedup.stderr
+++ b/tests/ui/single-use-lifetime/dedup.stderr
@@ -1,0 +1,11 @@
+error[E0392]: lifetime parameter `'a` is never used
+  --> $DIR/dedup.rs:6:12
+   |
+LL | struct Foo<'a>;
+   |            ^^ unused lifetime parameter
+   |
+   = help: consider removing `'a`, referring to it in a field, or using a marker such as `PhantomData`
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0392`.

